samples/add-features-with-contingent-values/src/main/java/com/esri/arcgismaps/sample/addfeatureswithcontingentvalues/components/MapViewModel.kt

```diff
@@ -0,0 +1,353 @@
+        feature?.attributes?.set("Protection", codedValue.code)
+        _featureEditState.value = _featureEditState.value.copy(
+            protection = codedValue,
+            bufferRange = featureTable?.bufferRange()
+        )
+    }
+
+    private fun ArcGISFeatureTable.bufferRange(): ContingentRangeValue? {
+        val contingentValueResult = feature?.let {
+            getContingentValuesOrNull(it, "BufferSize")
+        }
+
+        return contingentValueResult?.byFieldGroup?.get("BufferSizeFieldGroup")?.get(0) as? ContingentRangeValue
+    }
+
+    fun onBufferSizeSelected(bufferSize: Int) {
+        feature?.attributes?.set("BufferSize", bufferSize)
+        _featureEditState.value = _featureEditState.value.copy(buffer = bufferSize)
+    }
+
+    /**
+     * Ensure that the selected values are a valid combination.
+     * If contingencies are valid, then display [feature] on the [mapPoint]
+     */
+    fun validateContingency(mapPoint: Point) {
+        val resources = getApplication<Application>().resources
+        // check if all the features have been set
+        if (featureTable == null) {
+            messageDialogVM.showMessageDialog(resources.getString(R.string.input_all_values))
+            return
+        }
+
+        // validate the feature's contingencies
+        val contingencyViolations = feature?.let {
+            featureTable?.validateContingencyConstraints(it)
+        } ?: return messageDialogVM.showMessageDialog(resources.getString(R.string.no_feature_created))
+
+        // if there are no contingency violations the feature is valid and ready to add to the feature table
+        if (contingencyViolations.isEmpty()) {
+
+            // set the geometry of the feature to the map point
+            feature?.geometry = mapPoint
+
+            // create the buffer graphic for the feature
+            val bufferGraphic = feature?.let { createBufferGraphic(it) }
+
+            // add the graphic to the graphics overlay
+            bufferGraphic?.let { graphicsOverlay.graphics.add(it) }
+
+            // add the feature to the feature table
+            viewModelScope.launch {
+                feature?.let { featureTable?.addFeature(it) }
+            }
+        } else {
+            val violations = contingencyViolations.joinToString(separator = "\n") {
+                violation -> violation.fieldGroup.name
+            }
+            messageDialogVM.showMessageDialog(
+                "Invalid contingent values",
+                "${contingencyViolations.size} violations found:\n" + violations
+            )
+        }
+
+    }
+
+    /**
+     * Clears feature and attributes held in the view model to avoid inconsistent state
+     * after feature is created, fails to create, etc.
+     */
+    fun clearFeature() {
+        feature = null
+        _featureEditState.value = featureTable?.let {FeatureEditState(statusAttributes = it.statusFieldCodedValues())}
```

Comment:
nitpick - formatting:
```suggestion
        _featureEditState.value = featureTable?.let { FeatureEditState(statusAttributes = it.statusFieldCodedValues()) }
```